packages/server/src/client/internal/open-feature-client.ts

```diff
@@ -276,22 +277,47 @@ export class OpenFeatureClient implements Client {
 
     const mergedContext = this.mergeContexts(invocationContext);
 
-    // this reference cannot change during the course of evaluation
-    // it may be used as a key in WeakMaps
-    const hookContext: Readonly<HookContext> = {
-      flagKey,
-      defaultValue,
-      flagValueType: flagType,
-      clientMetadata: this.metadata,
-      providerMetadata: this._provider.metadata,
-      context: mergedContext,
-      logger: this._logger,
-    };
+    // Create hook data instances for each hook
+    const hookDataMap = new WeakMap<Hook, DefaultHookData>();
```

Comment:
I don't think a WeakMap is correct here as the only reference to the initial data would be in the weak map, which would allow it to be collected before invocation.

The collection itself is scoped do the duration of hook execution, so a normal map should be fine as we wouldn't be retaining any strong references longer than hook invocation.

I am not completely sure about indexing by the hook instance as it maybe would be somewhat unexpected if you registered the same hook multiple times.

Reply:
I think you're right. I probably can use a map in both instances because they'll both be GCed after the evaluation.